independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java

```diff
@@ -324,16 +324,23 @@ public Optional<ResourceClass> createEndpoints(ClassInfo classInfo, boolean cons
 
             return Optional.of(clazz);
         } catch (Exception e) {
-            if (Modifier.isInterface(classInfo.flags()) || Modifier.isAbstract(classInfo.flags())) {
-                //kinda bogus, but we just ignore failed interfaces for now
-                //they can have methods that are not valid until they are actually extended by a concrete type
-                log.debug("Ignoring interface " + classInfo.name(), e);
-                return Optional.empty();
+            if (!isSpringRestController(classInfo)) {
+                if (Modifier.isInterface(classInfo.flags()) || Modifier.isAbstract(classInfo.flags())) {
+                    //kinda bogus, but we just ignore failed interfaces for now
+                    //they can have methods that are not valid until they are actually extended by a concrete type
+                    log.debug("Ignoring interface " + classInfo.name(), e);
+                    return Optional.empty();
+                }
             }
             throw new RuntimeException(e);
         }
     }
 
+    private boolean isSpringRestController(ClassInfo classInfo) {
+        return classInfo
+                .declaredAnnotation(DotName.createSimple("org.springframework.web.bind.annotation.RestController")) != null;
+    }
+
```

Comment:
I don't really want to add any Spring relate stuff to the core modules of RESTEasy Reactive, so I suggest introducing some sort of SPI for this

Reply:
Ah yes, makes sense. 
We would need to add a build item to pass "predicates".